File: app/src/main/java/com/example/musclemate/Database/Entity/ExerciseDatabase.java
package com.example.musclemate.Database.Entity;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Exercises.class}, version = 1)
public abstract class ExerciseDatabase extends RoomDatabase {
    private static final String DB_NAME = "exercise_db";
    private static ExerciseDatabase instance;

    public static synchronized ExerciseDatabase getInstance(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(),
                    ExerciseDatabase.class, DB_NAME).allowMainThreadQueries().build();
        }
        return instance;
    }

    public abstract ExerciseDao exerciseDao();
}

File: app/src/main/java/com/example/musclemate/Database/exerciseDatabase.kt
package com.example.musclemate.Database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.musclemate.Database.Entity.Exercises

@Database(entities = [Exercises::class], version = 1)
abstract class ExerciseDatabase : RoomDatabase() {

    abstract fun getExerciseDao(): ExerciseDao

    companion object {
        @Volatile
        private var instance: ExerciseDatabase? = null
        private val LOCK = Any()

        operator fun invoke(context: Context) = instance ?: synchronized(LOCK) {
            instance ?: createDatabase(context).also {
                instance = it
            }
        }

        private fun createDatabase(context: Context) =
            Room.databaseBuilder(context.applicationContext, ExerciseDatabase::class.java, "ExerciseDatabase.db").build()
    }
}